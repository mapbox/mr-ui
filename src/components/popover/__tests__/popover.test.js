import React from 'react';
import { shallow } from 'enzyme';
import { noDisplayCases } from './popover-test-cases';

describe('Popover', () => {
  let testCase;
  let wrapper;

  describe(noDisplayCases.basic.description, () => {
    beforeEach(() => {
      testCase = noDisplayCases.basic;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(noDisplayCases.dark.description, () => {
    beforeEach(() => {
      testCase = noDisplayCases.dark;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(noDisplayCases.warning.description, () => {
    beforeEach(() => {
      testCase = noDisplayCases.warning;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(noDisplayCases.noPointer.description, () => {
    beforeEach(() => {
      testCase = noDisplayCases.noPointer;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(noDisplayCases.allProps.description, () => {
    beforeEach(() => {
      testCase = noDisplayCases.allProps;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
