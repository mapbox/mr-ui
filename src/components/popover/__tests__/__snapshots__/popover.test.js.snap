// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover all props renders 1`] = `
<Displaced
  alignment="top"
  allowPlacementAxisChange={false}
  containWithinViewport={false}
  getAnchorElement={[Function]}
  getContainingElement={[Function]}
  hasPointer={false}
  hideWhenAnchorIsOffscreen={true}
  mounted={true}
  onElement={[Function]}
  placement="left"
  pointerColor="#fff"
>
  <div
    className="color-text shadow-darken25 round"
    data-popover-id={4}
    data-test={true}
    key="body"
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="dialog"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
    tabIndex={-1}
  >
    <div
      className="w240 h240"
    >
      test popover
    </div>
  </div>
</Displaced>
`;

exports[`Popover basic renders 1`] = `
<Displaced
  alignment="top"
  allowPlacementAxisChange={true}
  containWithinViewport={true}
  getAnchorElement={[Function]}
  hasPointer={true}
  hideWhenAnchorIsOffscreen={false}
  mounted={true}
  onElement={[Function]}
  placement="right"
  pointerColor="#fff"
>
  <div
    className="color-text shadow-darken25 round px12 py12"
    data-popover-id={1}
    key="body"
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="dialog"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
    tabIndex={-1}
  >
    <div
      className="w240 h240"
    >
      test popover
    </div>
  </div>
</Displaced>
`;

exports[`Popover dark renders 1`] = `
<Displaced
  alignment="top"
  allowPlacementAxisChange={true}
  containWithinViewport={true}
  getAnchorElement={[Function]}
  hasPointer={true}
  hideWhenAnchorIsOffscreen={false}
  mounted={true}
  onElement={[Function]}
  placement="right"
  pointerColor="#273d56"
>
  <div
    className="color-white shadow-darken25 round px12 py12"
    data-popover-id={2}
    key="body"
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="dialog"
    style={
      Object {
        "backgroundColor": "#273d56",
      }
    }
    tabIndex={-1}
  >
    <div
      className="w240 h240"
    >
      test popover
    </div>
  </div>
</Displaced>
`;

exports[`Popover warning renders 1`] = `
<Displaced
  alignment="top"
  allowPlacementAxisChange={true}
  containWithinViewport={true}
  getAnchorElement={[Function]}
  hasPointer={true}
  hideWhenAnchorIsOffscreen={false}
  mounted={true}
  onElement={[Function]}
  placement="right"
  pointerColor="#FFF5A0"
>
  <div
    className="color-text shadow-darken25 round px12 py12"
    data-popover-id={3}
    key="body"
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="dialog"
    style={
      Object {
        "backgroundColor": "#FFF5A0",
      }
    }
    tabIndex={-1}
  >
    <div
      className="w240 h240"
    >
      test popover
    </div>
  </div>
</Displaced>
`;
